beaker/src/stamp_manager.rs

```diff
@@ -0,0 +1,218 @@
+//! Stamp file management for Make-compatible incremental builds
+//!
+//! This module handles the creation and management of stamp files that enable
+//! Make to perform accurate incremental builds. Each stamp file contains a
+//! deterministic hash based only on inputs that affect the byte-level output
+//! of that stage.
+
+use anyhow::{anyhow, Result};
+use sha2::{Digest, Sha256};
+use std::path::{Path, PathBuf};
+
+use crate::cache_common::calculate_md5;
+use beaker_stamp::write_cfg_stamp;
+
+#[cfg(test)]
+use std::fs;
+
+/// Generate tool version hash
+pub fn generate_tool_hash() -> String {
+    let mut hasher = Sha256::new();
+    hasher.update(format!("beaker:{}", env!("CARGO_PKG_VERSION")));
+    format!("{:x}", hasher.finalize())[..16].to_string()
+}
+
+/// Generate model hash from file path
+pub fn generate_model_hash_from_path(model_path: &Path) -> Result<String> {
+    if model_path.exists() {
+        calculate_md5(model_path).map(|hash| hash[..16].to_string())
+    } else {
+        Err(anyhow!("Model file not found: {}", model_path.display()))
+    }
+}
+
+/// Create stamp file if content has changed
+///
+/// This is a legacy function that duplicates beaker-stamp functionality.
```

Comment:
Delete this function then and update all callers. We don't need backwards compat or legacy code.